cmd/snap/cmd_run_test.go

```diff
@@ -1900,26 +1900,31 @@ func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlow(c *check.C) {
 	defer restoreIsGraphicalSession()
 
 	restoreTryNotifyRefresh := snaprun.MockTryNotifyRefreshViaSnapDesktopIntegrationFlow(func(snapName string) (bool, error) {
+		c.Check(snapName, check.Equals, "some-snap")
 		return false, nil
 	})
 	defer restoreTryNotifyRefresh()
 
-	var notification *usersessionclient.PendingSnapRefreshInfo
 	restorePendingRefreshNotification := snaprun.MockPendingRefreshNotification(func(refreshInfo *usersessionclient.PendingSnapRefreshInfo) error {
-		notification = refreshInfo
+		c.Check(refreshInfo, check.DeepEquals, &usersessionclient.PendingSnapRefreshInfo{
+			InstanceName:  "some-snap",
+			TimeRemaining: 0,
```

Comment:
This is already the default value. You can leave it if you want to be very explicit about what the check expects but then I guess the same would apply to the checks below

Reply:
[This check existed](https://github.com/snapcore/snapd/blob/master/cmd/snap/cmd_run_test.go#L1938). It was just reworked to perform check inside the mock instead of checking object `notification` at a later stage.

This check was repeated on similar situations in as my interpretation of what was suggested for [here](https://github.com/snapcore/snapd/pull/12535#discussion_r1091983700)

Note that `PendingSnapRefreshInfo` has a `TimeRemaining` field where `FinishedSnapRefreshInfo` does not.